### PR TITLE
docs: Add v1.3.4 release notes

### LINKED ISSUE #63: Add v1.3.4 release notes
Release `v1.3.4` should be the one that encompases all of @j-s-ashley's hard work and should be the last Python 3.7 and Python 3.8 release (right after `v1.3.4` a `v1.4.0` release that is Python 3.9+ should be made).

Release notes (https://energyflow.network/releases/) for `v1.3.4` need to be made in advance of `v1.3.4` getting cut.

For reasons unknown to me, `energyflow`'s docs are currently all in handwritten HTML. So the release notes need to get manually written out in `docs/releases/index.html` below

https://github.com/thaler-lab/EnergyFlow/blob/6e90ca9c078c90303360900435fab78769bcb8e5/docs/releases/index.html#L161

### PR DESCRIPTION
**Pull request by matthewfeickert**
Resolves #63 

* Release v1.3.4 covers the work done by Jordan Ashley in their 2024 IRIS-HEP Fellow project.
   - c.f. https://iris-hep.org/fellows/JordanAshley.html

### CODE REVIEW

**Comment by matthewfeickert** [comment]
@rikab @j-s-ashley this is ready for review.

**Comment by matthewfeickert** [comment]
For easier reviewing:

> <a href="https://github.com/thaler-lab/EnergyFlow/compare/v1.3.3...v1.3.4">
>   <p><strong>1.3.4</strong></p>
> </a>
> <ul>
>   <li>Removed dependency on <code>six</code>.</li>
>   <li>Used modern TensorFlow and Keras to support Python <code>3.7+</code>.</li>
>   <li>Require NumPy <code>v1.17.0+</code> to support recommended use of <code>numpy.random.default_rng</code> over <code>numpy.random.random</code>.</li>
>   <li>Added optional <code>rng</code> and <code>seed</code> arguments to <code>energyflow.gen_random_events</code> and <code>energyflow.gen_random_events_mcom</code> to allow passing of a <code>numpy.random.Generator</code>.</li>
>   <li>Adopted <code>hatchling</code> as the build backend.</li>
>   <li>Added support for <a href="https://numpy.org/doc/2.0/release/2.0.0-notes.html">NumPy <code>2.0</code></a>.</li>
>   <li>Added <a href="https://github.com/conda-forge/energyflow-feedstock">distribution on conda-forge</a>.</li>
> </ul>
